File: server/routes/face_registration.js
const express = require("express");
const Face = require("../models/face");

const faceRegistration = express.Router();

module.exports = faceRegistration;

faceRegistration.post("/api/faceIsRegistred", async (req, res) => {
  try {
    const { name } = req.body;
    const faceExists = await Face.findOne({ name });
    if (faceExists) {
      res.status(300).json({ msg: "already-registered" });
      return;
    }
    res.status(200).json({ msg: "does-not-exist" });
  } catch (e) {
    res.status(500).json({ error: e.message });
  }
});

faceRegistration.get("/api/get-faces", async (req, res) => {
  try {
    const faces = await Face.find({});
    res.json(faces);
  } catch (e) {
    res.status(500).json({ error: "something went wrong!" });
  }
});

faceRegistration.post("/api/delete-face", async (req, res) => {
  try {
    const { name } = req.body;
    const existingFace = await Face.findOne({ name });
    if (!existingFace) {
      res.status(300).json({ msg: "not-registered" });
      return;
    }
    await Face.findOneAndDelete({ name });
    res.json({ msg: "deleted" });
  } catch (e) {
    res.status(500).json({ error: "something went wrong!" });
  }
});

faceRegistration.post("/api/register-face", async (req, res) => {
  try {
    const { name, secureUrl } = req.body;
    let face = new Face({
      name,
      secureUrl,
    });
    face = await face.save();
    res.status(200).json({ msg: "added" });
  } catch (e) {
    res.status(500).json({ error: e.message });
    console.log(e.message);
  }
});

File: lib/features/home/services/face_registration.dart
import 'dart:convert';

import 'package:anomaly_detection_system/constants/global_variables.dart';
import 'package:anomaly_detection_system/constants/utils.dart';
import 'package:anomaly_detection_system/features/community/models/face.dart';
import 'package:cloudinary_public/cloudinary_public.dart';
import 'package:flutter/cupertino.dart';
import 'package:image_picker/image_picker.dart';
import 'package:http/http.dart' as http;

Future<void> unregister(
    {required BuildContext context,
    required String email,
    required String name}) async {
  final unregistrationDetails = await http.post(
    Uri.parse(
      '$uri/api/delete-face',
    ),
    headers: <String, String>{
      'Content-Type': 'application/json; charset=UTF-8',
    },
    body: Face(id: '', name: '$email$name', secureUrl: '').toJson(),
  );
  final result = jsonDecode(unregistrationDetails.body)['msg'];
  print("RESULT " + result.toString());
  if (result == 'deleted') {
    showSnackBar(context, "Face Successfully unregistered");
  } else {
    showSnackBar(context, "This face is not registered");
  }
}

Future<void> register(
    {required BuildContext context,
    required XFile image,
    required String name}) async {
  final registrationDetails = await http.post(
    Uri.parse(
      '$uri/api/faceIsRegistred',
    ),
    headers: <String, String>{
      'Content-Type': 'application/json; charset=UTF-8',
    },
    body: Face(id: '', name: name, secureUrl: '').toJson(),
  );
  final String result = jsonDecode(registrationDetails.body)['msg'];
  if (result == 'already-registered') {
    showSnackBar(context, 'Face Already Registered');
    return;
  }
  final cloudinary = CloudinaryPublic(
    'dkktq9syf',
    'qqtfkrf4',
  );

  final response = await cloudinary.uploadFile(
    CloudinaryFile.fromFile(
      image.path,
      folder: name,
    ),
  );
  final secureUrl = response.secureUrl;
  print('BEFORE');
  await http.post(
    Uri.parse('$uri/api/register-face'),
    headers: <String, String>{
      'Content-Type': 'application/json; charset=UTF-8',
    },
    body: Face(id: '', name: name, secureUrl: secureUrl).toJson(),
  );
  print('AFTER');
  showSnackBar(context, 'Your Face is successfully registered');
  return;
}
